src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -215,12 +228,18 @@ def register_extra_cli_argument(command, dest, **kwargs):
     '''
     _cli_extra_argument_registry[command][dest] = CliCommandArgument(dest, **kwargs)
 
+def cli_command_with_handler(module_name, name, operation, client_factory=None, transform=None, table_transformer=None):
+    """ Registers a default Azure CLI command. These commands require no special parameters. """
+    command_table[name] = create_command(module_name, name, operation, transform, table_transformer,
+                                         client_factory, op_lazy=True)
+
+# TODO-DEREK Deprecate this method
 def cli_command(name, operation, client_factory=None, transform=None, table_transformer=None):
     """ Registers a default Azure CLI command. These commands require no special parameters. """
-    command_table[name] = create_command(name, operation, transform, table_transformer,
-                                         client_factory)
+    command_table[name] = create_command(None, name, operation, transform, table_transformer,
+                                         client_factory, op_lazy=False)
 
-def create_command(name, operation, transform_result, table_transformer, client_factory):
+def create_command(module_name, name, operation, transform_result, table_transformer, client_factory, op_lazy=False):
```

Comment:
Is op_lazy a temporary variable that will go away once the transition is complete?